torch/csrc/jit/codegen/cuda/evaluator_common.cpp

```diff
@@ -290,52 +294,62 @@ void NaiveIntegerMachine<IRContext>::runInstruction(int index) {
 }
 
 template <typename IRContext>
-void NaiveIntegerMachine<IRContext>::runUnaryOp(int index) {
+void NaiveValueMachine<IRContext>::runUnaryOp(int index) {
   int src_index = src0_[index];
-  bool src_defined = precomputed_integers_.defined_[src_index];
-  bool src_is_const = precomputed_integers_.is_constant_[src_index];
+  bool src_defined = precomputed_values_.defined_[src_index];
+  bool src_is_const = precomputed_values_.is_constant_[src_index];
   if (!src_defined && !src_is_const) {
     return;
   }
 
   int dest_index = dest_[index];
 
-  auto& src = precomputed_integers_.values_[src_index];
-  auto& dest = precomputed_integers_.values_[dest_index];
+  auto& src = precomputed_values_.values_[src_index];
+  auto& dest = precomputed_values_.values_[dest_index];
 
   switch (uop_type_[index]) {
     case UnaryOpType::Neg:
       dest = -src;
       break;
-    case UnaryOpType::Cast:
+    case UnaryOpType::Set:
       dest = src;
       break;
+    case UnaryOpType::Cast:
+      if (data_type_[index] == DataType::Double) {
+        dest = src.template cast<double>();
+      } else if (data_type_[index] == DataType::Int) {
+        dest = src.template cast<int64_t>();
+      } else {
+        TORCH_INTERNAL_ASSERT(false, "wrong type");
```

Comment:
NIT: Could you put something more along the lines of `dtype not supported in evaluator`?

Reply:
fixed